vm/actor/src/builtin/puppet/mod.rs

```diff
@@ -41,7 +41,21 @@ pub struct SendReturn {
 #[derive(Default, Serialize, Deserialize)]
 pub struct FailToMarshalCBOR {}
 
-impl Cbor for FailToMarshalCBOR {}
+impl Cbor for FailToMarshalCBOR {
+    fn marshal_cbor(&self) -> Result<Vec<u8>, Error> {
+        Err(Error::Marshalling {
+            description: "Automatic fail to Marshall".to_string(),
+            protocol: CodecProtocol::Cbor,
+        })
+    }
+
+    fn unmarshal_cbor(_bz: &[u8]) -> Result<Self, Error> {
+        Err(Error::Unmarshalling {
+            description: "Automatic fail to Unmarshal".to_string(),
+            protocol: CodecProtocol::Cbor,
+        })
+    }
+}
```

Comment:
can you instead just have the serialize and deserialize impls fail? I'm pretty sure the cbor trait is used for all use cases, but there might be some serializations using the serde trait, and that would pass (just trying to future proof) the cbor implementation uses the serde impls by the way so this can be removed.